circuit-test/src/commonMain/kotlin/com/slack/circuit/test/CircuitReceiveTurbine.kt

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2024 Slack Technologies, LLC
+// SPDX-License-Identifier: Apache-2.0
+package com.slack.circuit.test
+
+import app.cash.turbine.Event
+import app.cash.turbine.ReceiveTurbine
+import com.slack.circuit.runtime.CircuitUiState
+
+/**
+ * A Circuit-specific extension to [ReceiveTurbine] with extra helper functions for Circuit testing.
+ */
+public interface CircuitReceiveTurbine<UiState : CircuitUiState> : ReceiveTurbine<UiState> {
+  /** Awaits the _next_ [UiState] emission that is different than the [lastState]. */
+  public suspend fun consumeState(lastState: UiState? = tryMostRecentItem()): UiState
```

Comment:
This is an interesting approach. Not sure it needs to be a public param? What gets unblocked with that?

Reply:
thinking that in some cases you've already held on to the previous state, but we can always close it off and open it up later

Reply:
Removed!